src/realm/error_codes.cpp

```diff
@@ -119,40 +119,29 @@ ErrorCategory ErrorCodes::error_categories(Error code)
             return ErrorCategory().set(ErrorCategory::invalid_argument).set(ErrorCategory::logic_error);
 
         case CustomError:
-            return ErrorCategory()
-                .set(ErrorCategory::runtime_error)
-                .set(ErrorCategory::app_error)
-                .set(ErrorCategory::custom_error);
+            return ErrorCategory().set(ErrorCategory::app_error).set(ErrorCategory::custom_error);
```

Comment:
Sorry, I must have not been clear, in my past comment I was suggesting just not changing any of the existing error categories or which codes fall into what but making the AppError inherit from Exception so it doesn't have to only be RuntimeErrors.

Reply:
Sorry, I thought I was reverting the changes back. I'll fix these.